### PR TITLE
Fix DistributedRectilinearMeshGenerator unique_id

### LINKED ISSUE #14997: Fix unique_id generation in MOOSE mesh generators
## Bug Description
A few of the MOOSE CI tests failing with https://github.com/libMesh/libmesh/pull/2491 appear at first inspection to be due to conflicts between the MOOSE DistributedRectilinearMeshGenerator unique_id assignment and the new libMesh assertions (specifically that node and elem unique_id assignments do not overlap).  Also, looking at the code idiom used for unique_id assignment in that mesh generator, I think I see a bug (similar to some bugs just fixed in libMesh in that PR) that will trigger if mesh generation is followed by any automatic unique_id assignment e.g. during mesh refinement.

## Steps to Reproduce
For the first problem, run any of the DistributedRectilinearMeshGenerator tests using the updated libmesh_assert_valid_unique_ids() test from https://github.com/libMesh/libmesh/pull/2491

For the second, do the same, but with refinement steps following mesh generation.

## Impact
Urgent; the CI failures prevent me from fully merging https://github.com/libMesh/libmesh/pull/2491

The node/elem unique_id overlap hopefully isn't something people are relying on not seeing.

The bug with subsequent mesh refinement could affect anyone relying on unique_id with this mesh generator.

### PR DESCRIPTION
**Pull request by roystgnr**
This is the only fix in MOOSE itself that my tests needed - all the other failures in https://github.com/libMesh/libmesh/pull/2491 I was able to track down to underlying issues remaining in libMesh.

Closes #14997 - I'll open a new ticket if the CI for modules or apps or dependent tests or any such thing manages to trigger further assertion failures.

I don't really *like* the idiom I'm using here, but it's the most flexible and it doesn't depend on https://github.com/libMesh/libmesh/pull/2491 to work.  After that libMesh PR makes it downstream we might shave off a few lines of code here by using an API upgrade I'm putting in the commits there.

### CODE REVIEW

File: framework/src/meshgenerators/DistributedRectilinearMeshGenerator.C [hunk 1]
```diff
@@ -216,14 +216,29 @@ DistributedRectilinearMeshGenerator::addElement<Edge2>(const dof_id_type nx,
 
   auto node_offset = elem_id;
 
-  auto node0_ptr = mesh.add_point(Point(static_cast<Real>(node_offset) / nx, 0, 0), node_offset);
-  node0_ptr->set_unique_id() = node_offset;
-  node0_ptr->processor_id() = pid;
+  Node * node0_ptr = mesh.query_node_ptr(node_offset);
```
Comment (fdkong) [inline framework/src/meshgenerators/DistributedRectilinearMeshGenerator.C:219]: Does this query involve communication?

**Review by fdkong — CHANGES_REQUESTED** [review]

**Comment by fdkong** [comment]
@roystgnr It looks fine for me. But I do have a query

File: framework/src/meshgenerators/DistributedRectilinearMeshGenerator.C (hunk 1 above)
Comment (roystgnr) [reply framework/src/meshgenerators/DistributedRectilinearMeshGenerator.C:219]: No; this the "see if a object with the requested id exists" API, purely local.

**Review by fdkong — APPROVED** [review]

**Comment by fdkong** [comment]
Thanks to @roystgnr

**Pull request merged**